### PR TITLE
move go:generate calls to scripts/generate.sh

### PR DESCRIPTION
**Pull request by peter-edge**
Resolves https://github.com/yarpc/yarpc-go/issues/627

I like to just put the code up so you can see what it would look like, and it was a quick task, feel free to reject if you don't like this though.

I think this cleans up a bunch though. For example, I'm betting some of those thrift calls, when put next to each other, might be changed to thriftrw, and stringer was always failing so now we see it. Also, the license generation is done once (and we could remove the scripts/generateLicenses.sh file too).

There's some other stuff, namely the dependencies, that need to be added, but I started this on another PR. Really, all the building of the dependencies could be brought out of the Makefile and done here, which might make the Makefile a bit easier heh.

### CODE REVIEW

**Comment by breerly** [comment]
We should also verify there are no occurrences of `//go:generate`, and if so, fail the build.

**Comment by peter-edge** [comment]
Will do, good check. I'd argue generally (across all projects) to have a
set of linters that are run in parallel we can add to.

**Comment by peter-edge** [comment]
Should we add such a check to the make lint target, the generate script, or
just for CI? I'd argue for the script.

**Comment by breerly** [comment]
@peter-edge trust your gut - just make sure it runs during the build :)

**Comment by peter-edge** [comment]
There's more nerve endings in my gut than my brain, so I will (let me know
if that joke doesn't make sense, will send YouTube link :) )

**Comment by abhinav** [comment]
I'm not against this change but can you make the generate.sh more readable?

**Comment by peter-edge** [comment]
What should I do in your opinion? I just see it as a list of generate commands, not sure what we could do to clean it up:

- bash functions for each type of generate? eh
- a go program that does all this?

**Comment by peter-edge** [comment]
I added a `make nogogenerate` target and added it as a dependency of lint

File: Makefile [hunk 1]
```diff
@@ -66,8 +66,13 @@ build:
...
 .PHONY: generate
 generate: $(_GENERATE_DEPS_EXECUTABLES)
-	PATH=$(_GENERATE_DEPS_DIR):$$PATH go generate $(PACKAGES)
-	./scripts/updateLicenses.sh
+	PATH=$(_GENERATE_DEPS_DIR):$$PATH ./scripts/generate.sh
+
+.PHONY: nogogenerate
+nogogenerate:
+	$(eval NOGOGENERATE_LOG := $(shell mktemp -t nogogenerate.XXXXX))
+	@grep -n -R \/\/go:generate $(GO_FILES) 2>&1 > $(NOGOGENERATE_LOG) || true
```
Comment (abhinav) [inline Makefile]: Do you need `-R`? Isn't GO_FILES a list of files?

Comment (peter-edge) [inline Makefile]: whoops, yep

**Comment by peter-edge** [comment]
I'm not sure putting comments like:

```
# mockgen
mockgen ...
```

are helpful, but I split them into groups.

**Comment by peter-edge** [comment]
I believe this is OK now. I think the merge is proper, and I fixed the failing `stringer` command.

**Pull request merged**